go/private/actions/link.bzl

```diff
@@ -96,16 +96,18 @@ def emit_link(
         tool_args.add("-msan")
     if ((go.mode.static and not go.mode.pure) or
         go.mode.link != LINKMODE_NORMAL or
-        go.mode.goos == "windows" and (go.mode.race or go.mode.msan)):
+        go.mode.race or
+        go.mode.goos == "windows" and go.mode.msan):
         # Force external linking for the following conditions:
         # * Mode is static but not pure: -static must be passed to the C
         #   linker if the binary contains cgo code. See #2168, #2216.
         # * Non-normal build mode: may not be strictly necessary, especially
         #   for modes like "pie".
-        # * Race or msan build for Windows: Go linker has pairwise
-        #   incompatibilities with mingw, and we get link errors in race mode.
-        #   Using the C linker avoids that. Race and msan always require a
-        #   a C toolchain. See #2614.
+        # * Msan build for Windows: Go linker has pairwise incompatibilities
+        #   with mingw. Using the C linker avoids that. Race and msan always
+        #   require a C toolchain. See #2614.
+        # * Race builds (previously Windows, later +linux): we get linker
```

Comment:
Let's document the current state only, not the history of this part of the code. We can always rely on git history for that. 

Please also add an example of the error you get with the internal linker.

Reply:
Done, both suggestions.